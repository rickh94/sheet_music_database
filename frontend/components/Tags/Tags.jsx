import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withAlert } from 'react-alert'
import { Link } from 'react-router-dom'

import Container from 'react-bulma-components/lib/components/container/container'
import Card from 'react-bulma-components/lib/components/card/card'
import Heading from 'react-bulma-components/lib/components/heading'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Level from 'react-bulma-components/lib/components/level'

import Header from '../Header'
import FieldDisplay from '../FieldDisplay'
import { tags } from '../../actions'
import alertText, { messages } from '../../middleware/alertText'

export class Tags extends Component {
  state = {
    errors: {}
    // errors are saved as [id]: value
  }

  static propTypes = {
    token: PropTypes.string,
    history: PropTypes.object.isRequired,
    alert: PropTypes.object.isRequired,
    tags: PropTypes.object.isRequired,
    getTags: PropTypes.func.isRequired
  }

  componentDidMount() {
    if (!this.props.token) {
      this.props.alert.show(messages.notLoggedIn, { type: 'error' })
      this.props.history.push('/login')
    } else {
      const success = this.props.getTags(this.props.token)
      console.log(success)
      if (!success) {
        this.props.alert.show(...messages.couldNotGet('Tags'))
      }
    }
  }

  render() {
    const tagList = this.props.tags.list
    return (
      <React.Fragment>
        <Header />
        <Container>
          <Card className="margin-default">
            <Card.Content>
              <Heading size={3}>Tags</Heading>
              <ul>
                {tagList.map(tag => (
                  <TagItem
                    tag={tag}
                    key={tag.id}
                    saveCallback={() => {}}
                    deleteCallback={() => {}}
                    errors={this.state.errors[tag.id]}
                  />
                ))}
              </ul>
            </Card.Content>
          </Card>
        </Container>
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => {
  return {
    token: state.account.token,
    tags: state.tags
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getTags: token => {
      return dispatch(tags.getTags(token))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withAlert(Tags))

export function TagItem(props) {
  return (
    <Level>
      <FieldDisplay
        value={props.tag.name}
        saveCallback={props.saveCallback}
        backendFieldName=""
        errors={props.error}
        linkTo={`/tag/${props.tag.id}`}
      />
      <div className="links">
        <a
          onClick={e => {
            e.preventDefault()
            props.deleteCallback(props.tag.id)
          }}
          className="edit-link"
        >
          delete
        </a>
      </div>
    </Level>
  )
}

TagItem.propTypes = {
  tag: PropTypes.object.isRequired,
  saveCallback: PropTypes.func.isRequired,
  deleteCallback: PropTypes.func.isRequired,
  errors: PropTypes.string
}
