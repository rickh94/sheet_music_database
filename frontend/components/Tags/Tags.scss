.tag-header {
  margin: 0 !important;
}
