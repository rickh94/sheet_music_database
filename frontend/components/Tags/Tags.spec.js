import React from 'react'
import { shallow } from 'enzyme'
import { Tags, TagItem } from './Tags'
import '../../setupTests'

describe('Tags', () => {
  const testToken = 'testtoken'
  const wrapper = shallow(
    <Tags
      history={{ push: jest.fn() }}
      token={testToken}
      alert={{ show: jest.fn() }}
      getTags={jest.fn()}
      tags={{}}
    />
  )
  
  beforeEach(() => {
    wrapper.update()
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the header', () => {
    expect(wrapper.exists('Connect(Header)')).toBeTruthy()
  })

  it('renders a list', () => {
    expect(wrapper.exists('ul')).toBeTruthy()
  })

  it('renders tags from the list', () => {
    const tags = { list: [{ id: 1, name: 'test1' }, { id: 2, name: 'test2' }] }
    wrapper.setState({ tags })
    expect(wrapper.find('TagItem').length).toEqual(2)
  })

  it('redirects if not logged in', () => {
    const history = { push: jest.fn() }
    const token = null
    const alert = { show: jest.fn() }
    wrapper.setProps({ token, history, alert })
    wrapper.instance().componentDidMount()
    expect(history.push).toHaveBeenCalledWith('/login')
    expect(alert.show).toHaveBeenCalledWith(
      <span className="alert-text">Please log in</span>,
      { type: 'error' }
    )
  })

  it('gets tags if logged in', () => {
    const getTags = jest.fn()
    wrapper.setProps({ getTags, token: testToken })
    wrapper.instance().componentDidMount()
    expect(getTags).toHaveBeenCalledWith(testToken)
  })

  it('shows an error on get failure', () => {
    const getTags = () => false
    const alert = {show: jest.fn()}
    wrapper.setProps({getTags, token: testToken, alert})
    wrapper.instance().componentDidMount()
    expect(alert.show).toHaveBeenCalledWith(
      <span className="alert-text">Could not get Tags</span>,
      { type: 'error' }
    )
  })
})

describe('TagItem', () => {
  const testTag = { id: 5, name: 'test' }
  const wrapper = shallow(
    <TagItem tag={testTag} saveCallback={jest.fn()} deleteCallback={jest.fn()} />
  )

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a list item', () => {
    expect(wrapper.exists('li')).toBeTruthy()
  })

  it('has delete link with icon', () => {
    expect(wrapper.contains('delete')).toBeTruthy()
    expect(wrapper.find('FontAwesomeIcon')).toBeTruthy()
  })

  it('calls passed in function when delete is clicked', () => {
    const deleteCallback = jest.fn()
    wrapper.setProps({ deleteCallback })
    wrapper
      .findWhere(el => el.name() == 'a' && el.contains('delete'))
      .simulate('click', { preventDefault: jest.fn() })
    expect(deleteCallback).toHaveBeenCalledWith(testTag.id)
  })
})
