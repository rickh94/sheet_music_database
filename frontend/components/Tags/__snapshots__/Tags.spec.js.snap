// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagItem matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagItem
    deleteCallback={[MockFunction]}
    saveCallback={[MockFunction]}
    tag={
      Object {
        "id": 5,
        "name": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FieldDisplay
          backendFieldName=""
          linkTo="/tag/5"
          saveCallback={[MockFunction]}
          value="test"
        />,
        <div
          className="links"
        >
          <a
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="trash"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            delete
          </a>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backendFieldName": "",
          "errors": undefined,
          "linkTo": "/tag/5",
          "saveCallback": [MockFunction],
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="trash"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            delete
          </a>,
          "className": "links",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="trash"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "delete",
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "trash",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "delete",
          ],
          "type": "a",
        },
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FieldDisplay
            backendFieldName=""
            linkTo="/tag/5"
            saveCallback={[MockFunction]}
            value="test"
          />,
          <div
            className="links"
          >
            <a
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="trash"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              delete
            </a>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backendFieldName": "",
            "errors": undefined,
            "linkTo": "/tag/5",
            "saveCallback": [MockFunction],
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="trash"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              delete
            </a>,
            "className": "links",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="trash"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "delete",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "trash",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "delete",
            ],
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Tags matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    alert={
      Object {
        "show": [MockFunction] {
          "calls": Array [
            Array [
              <span
                className="alert-text"
              >
                Could not get Tags
              </span>,
              Object {
                "type": "error",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    getTags={
      [MockFunction] {
        "calls": Array [
          Array [
            "testtoken",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    tags={Object {}}
    token="testtoken"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Container
          breakpoint={null}
          fluid={false}
          renderAs="div"
          style={Object {}}
        >
          <Card
            className="margin-default"
            renderAs="div"
            style={Object {}}
          >
            <CardContent
              renderAs="div"
            >
              <Heading
                className=""
                heading={false}
                renderAs="h1"
                size={3}
                spaced={false}
                subtitle={false}
                weight={null}
              >
                Tags
              </Heading>
              <ul />
            </CardContent>
          </Card>
        </Container>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "breakpoint": null,
          "children": <Card
            className="margin-default"
            renderAs="div"
            style={Object {}}
          >
            <CardContent
              renderAs="div"
            >
              <Heading
                className=""
                heading={false}
                renderAs="h1"
                size={3}
                spaced={false}
                subtitle={false}
                weight={null}
              >
                Tags
              </Heading>
              <ul />
            </CardContent>
          </Card>,
          "className": undefined,
          "clearfix": undefined,
          "clipped": undefined,
          "fluid": false,
          "hidden": undefined,
          "invisible": undefined,
          "italic": undefined,
          "marginless": undefined,
          "overlay": undefined,
          "paddingless": undefined,
          "pull": undefined,
          "radiusless": undefined,
          "renderAs": "div",
          "responsive": undefined,
          "shadowless": undefined,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": undefined,
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": <CardContent
              renderAs="div"
            >
              <Heading
                className=""
                heading={false}
                renderAs="h1"
                size={3}
                spaced={false}
                subtitle={false}
                weight={null}
              >
                Tags
              </Heading>
              <ul />
            </CardContent>,
            "className": "margin-default",
            "clearfix": undefined,
            "clipped": undefined,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "children": Array [
                <Heading
                  className=""
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  Tags
                </Heading>,
                <ul />,
              ],
              "className": undefined,
              "clearfix": undefined,
              "clipped": undefined,
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": undefined,
              "overlay": undefined,
              "paddingless": undefined,
              "pull": undefined,
              "radiusless": undefined,
              "renderAs": "div",
              "responsive": undefined,
              "shadowless": undefined,
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "children": "Tags",
                  "className": "",
                  "clearfix": undefined,
                  "clipped": undefined,
                  "heading": false,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": undefined,
                  "overlay": undefined,
                  "paddingless": undefined,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "h1",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "size": 3,
                  "spaced": false,
                  "subtitle": false,
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                  "weight": null,
                },
                "ref": null,
                "rendered": "Tags",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "ul",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Container
            breakpoint={null}
            fluid={false}
            renderAs="div"
            style={Object {}}
          >
            <Card
              className="margin-default"
              renderAs="div"
              style={Object {}}
            >
              <CardContent
                renderAs="div"
              >
                <Heading
                  className=""
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  Tags
                </Heading>
                <ul />
              </CardContent>
            </Card>
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "breakpoint": null,
            "children": <Card
              className="margin-default"
              renderAs="div"
              style={Object {}}
            >
              <CardContent
                renderAs="div"
              >
                <Heading
                  className=""
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  Tags
                </Heading>
                <ul />
              </CardContent>
            </Card>,
            "className": undefined,
            "clearfix": undefined,
            "clipped": undefined,
            "fluid": false,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "children": <CardContent
                renderAs="div"
              >
                <Heading
                  className=""
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  Tags
                </Heading>
                <ul />
              </CardContent>,
              "className": "margin-default",
              "clearfix": undefined,
              "clipped": undefined,
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": undefined,
              "overlay": undefined,
              "paddingless": undefined,
              "pull": undefined,
              "radiusless": undefined,
              "renderAs": "div",
              "responsive": undefined,
              "shadowless": undefined,
              "style": Object {},
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": undefined,
                "children": Array [
                  <Heading
                    className=""
                    heading={false}
                    renderAs="h1"
                    size={3}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    Tags
                  </Heading>,
                  <ul />,
                ],
                "className": undefined,
                "clearfix": undefined,
                "clipped": undefined,
                "hidden": undefined,
                "invisible": undefined,
                "italic": undefined,
                "marginless": undefined,
                "overlay": undefined,
                "paddingless": undefined,
                "pull": undefined,
                "radiusless": undefined,
                "renderAs": "div",
                "responsive": undefined,
                "shadowless": undefined,
                "textAlignment": undefined,
                "textColor": undefined,
                "textSize": undefined,
                "textTransform": undefined,
                "textWeight": undefined,
                "unselectable": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "backgroundColor": undefined,
                    "children": "Tags",
                    "className": "",
                    "clearfix": undefined,
                    "clipped": undefined,
                    "heading": false,
                    "hidden": undefined,
                    "invisible": undefined,
                    "italic": undefined,
                    "marginless": undefined,
                    "overlay": undefined,
                    "paddingless": undefined,
                    "pull": undefined,
                    "radiusless": undefined,
                    "renderAs": "h1",
                    "responsive": undefined,
                    "shadowless": undefined,
                    "size": 3,
                    "spaced": false,
                    "subtitle": false,
                    "textAlignment": undefined,
                    "textColor": undefined,
                    "textSize": undefined,
                    "textTransform": undefined,
                    "textWeight": undefined,
                    "unselectable": undefined,
                    "weight": null,
                  },
                  "ref": null,
                  "rendered": "Tags",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "ul",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
